### ISSUE TITLE
[v1.34.0] Maestro unable to communicate with iOS driver when executed with node (Azure CI)

### ISSUE THREAD

**Issue by TimIsThinking** [mobile-dev-inc/maestro#1577]
**Describe the bug**
Executing Maestro using a node to execute a shell script results in Maestro being unable to communicate with the iOS driver.
This is how Azure CI runs scripts on it's agents. I have been able to replicate this scenario on my machine below.

**To Reproduce**
1. Start the simulator
2. Install your app (not required)
3. Run the node script `node runMaestroNode.js`
4. Notice that the iOS driver gets installed / uninstalled multiple times on the simulator
5. Notice that after timing out the script fails with `MaestroDriverStartupException$IOSDriverTimeoutException: Maestro iOS driver did not start up in time`
6. Notice that running the shell script directly instead `sh runMaestro.sh` works as expected

**Expected behavior**
Expect the script to start maestro, connect to iOS driver and begin running flows - as it does when running the shell script directly without node `sh runMaestro.sh`

**Environment information**
- Maestro version: v1.34.0
- Platform: iOS 16.2 & iOS 17 Simulators
- Framework: React Native
- Device model and OS version: iPhone 14 (MacOS 16.2 & 17)
- Simulator
- Host: Mac M1 (13.6.1) Xcode 14.2

**Workspace**
Using maestro samples

Shell script (runMaestro.sh)
```
#! /bin/bash
export MAESTRO_DRIVER_STARTUP_TIMEOUT=60000
$HOME/.maestro/bin/maestro -v
$HOME/.maestro/bin/maestro test samples/ios-flow.yaml
```

Node script (runMaestroNode.js)
```
const { exec } = require('child_process');

exec('sh runMaestro.sh', (error, stdout, stderr) => {
  if (error) {
    console.log(`error: ${error.message}`);
    return;
  }
  if (stderr) {
    console.log(`stderr: ${stderr}`);
  }
  console.log(`stdout: ${stdout}`);
});
```

**Bugreport output**
[maestro.log](https://github.com/mobile-dev-inc/maestro/files/13279242/maestro.log)

**Additional context**
Using `sudo tcpdump -v --interface lo0 port 22087` and running the shell script I can see many connections appear as expected, but when running the node script it is completely empty. Seems that there is something blocking maestro from making the requests in the first place?

https://mobile-dev-inc.slack.com/archives/C041FU72T54/p1699294395433169

Any suggestions would be much appreciated :)

**Comment by TimIsThinking** [comment]
After further debugging I discovered that this is a bug in JDK versions 11 & 12. In my basic understanding the bug drops support for IPv6 connections when stdin is open.

https://bugs.openjdk.org/browse/JDK-8211842
https://stackoverflow.com/a/52573455

After applying a workaround found here https://stackoverflow.com/a/38887598 (adding `< /dev/null` after shell script execution) I was able to get Maestro running in our self hosted mac os agent, running on Azure pipelines! 

Fix for the above example
Node script (runMaestroNode.js)
```
const { exec } = require('child_process');

exec('sh runMaestro.sh < /dev/null', (error, stdout, stderr) => {
  if (error) {
    console.log(`error: ${error.message}`);
    return;
  }
  if (stderr) {
    console.log(`stderr: ${stderr}`);
  }
  console.log(`stdout: ${stdout}`);
});
```
